poh-bundle-linux-x32/oh.sh

```diff
@@ -1,68 +1,271 @@
 
 MYSQL_DIR="mysql-5.7.30-linux-glibc2.12-i686"
-JAVA_DIR="zulu11.39.15-ca-jre11.0.7-linux_i686"
+MYSQL_URL="https://dev.mysql.com/get/Downloads/MySQL-5.7/"
+
+### JRE 8 - openlogic
+#JAVA_DISTRO="openlogic-openjdk-8u262-b10-linux-x32"
+#JAVA_URL="https://builds.openlogic.com/downloadJDK/openlogic-openjdk/8u262-b10/"
+#JAVA_DIR="openlogic-openjdk-8u262-b10-linux-32"
+
+### JRE 8 - zulu
+#JAVA_DISTRO="zulu8.50.0.21-ca-jre8.0.272-linux_i686"
+#JAVA_URL="https://cdn.azul.com/zulu/bin/"
+#JAVA_DIR="zulu8.50.0.21-ca-jre8.0.272-linux_i686"
+
+### JRE 11
+JAVA_DISTRO="zulu11.43.21-ca-jre11.0.9-linux_i686"
+JAVA_URL="https://cdn.azul.com/zulu/bin/"
+JAVA_DIR="zulu11.43.21-ca-jre11.0.9-linux_i686"
+
+# Database
+MYSQL_PORT=3307
+MYSQL_SOCKET="var/run/mysqld/mysql.sock"
+DB_CREATE_SQL="database.sql"
+DB_ARCHIVED_SQL="database.sql.imported"
+
 OH_DIR="oh"
+DATABASE_NAME="oh"
+DATABASE_USER="isf"
+DATABASE_PASSWORD="isf123"
 DICOM_DEFAULT_SIZE="4M"
 
-where_i_am=$(pwd)
-cd $(dirname $0)
-POH_PATH=$(pwd)
-cd $where_i_am
+######################## DO NOT EDIT BELOW THIS LINE ########################
+# name of this shell script
+SCRIPT_NAME=$(basename "$0")
+
+# set current dir
+CURRENT_DIR=$PWD
+
+# check user
+if [ $(id -u) -eq 0 ]
+  then echo "Error - do not run this script as root."
+  exit 1
+fi
+
+# set POH_PATH if not defined
+if [ -z $POH_PATH ]; then
+	echo "Warning - POH_PATH not found. Please set it up properly."
+		if [ ! -f ./$SCRIPT_NAME ]; then
+		echo "Error - oh.sh not found in the current PATH. Please cd the directory where POH was unzipped or set up POH_PATH properly."
+		exit 0
+		fi
+	POH_PATH=$PWD
+fi
+
+cd $POH_PATH
+
+######## User input / option parsing
+
+function script_usage {
+	echo "Usage: $(basename $0) [-h -r]" 2>&1
+        echo "   -h       shows this short help"
+        echo "   -r       reset Portable Open Hospital installation"
+	exit 0
+}
+
 
-# Find a free port to run MySQL starting from the default port.
-mysql_port=3306
-while [ $(ss -tna | awk '{ print $4 }' | grep ":$mysql_port") ]; do
-	mysql_port=$(expr $mysql_port + 1)
+function restore_db {
+	if [ -f $POH_PATH/$DB_CREATE_SQL ];
+	then
+	        echo "Found SQL creation script, starting OH...."
+
+	elif [ -f $POH_PATH/$DB_ARCHIVED_SQL ];
+	then
+	        echo "Found archived SQL creation script, restoring and starting OH...."
+		mv $POH_PATH/$DB_ARCHIVED_SQL $POH_PATH/$DB_CREATE_SQL
+	fi
+}
+
+function start_database {
+	cd $POH_PATH/$MYSQL_DIR/
+	echo "Starting MySQL... "
+	$POH_PATH/$MYSQL_DIR/bin/mysqld_safe --defaults-file=$POH_PATH/etc/mysql/my.cnf 2>&1 > /dev/null &
+	if [ $? -ne 0 ]; then
+		echo "Error: Database not started!"
+		exit 2
+	fi
+}
+
+
+# list of arguments expected in the input
+optstring=":hr"
+
+# function to parse input
+while getopts ${optstring} arg; do
+	case ${arg} in
+	"h")
+		script_usage
+		;;
+	"r")
+        	echo "Resetting Portable Open Hospital installation...."
+		restore_db
```

Comment:
so these optional functions are meant to be executed alternatively to the normal execution or additionally, before it? 'Cause it looks like it would restore the db and then execute OH - just checking that that's the intended behaviour.

Reply:
Alternatively, so restore_db is called only if you want the reset the installation. Maybe the name of the function is not optimal :-)

Reply:
then I guess there should be an `exit` after restore_db, if I understand correctly...?

Reply:
Sorry @pviotti, I completely misunderstood your first comment.
The restore_db is invoked __additionaly__ before the normal run with the -r option: if it finds the archived copy of database.sql, restores it and then proceed with the normal startup and execution of OH).
It should make sense, hopefully :-)